framework/topology.cpp

```diff
@@ -774,9 +774,23 @@ void apply_cpuset_param(char *param)
     LogicalProcessorSet result = {};
     new_cpu_info.reserve(old_cpu_info.size());
 
+    bool add = true;
+    if (*param == '!') {
+        // we're removing from the existing set
+        new_cpu_info = { old_cpu_info.begin(), old_cpu_info.end() };
+        add = false;
+        ++param;
+    }
+
     std::string p = param;
     for (char *arg = strtok(p.data(), ","); arg; arg = strtok(nullptr, ",")) {
         const char *orig_arg = arg;
+        struct MatchCpuInfoByCpuNumber {
```

Comment:
struct declaration in a loop looks weird, would be better to move it out or event to the beginning of the function.

Reply:
Moved.